app/layersmodel.cpp

```diff
@@ -192,6 +213,11 @@ int LayersModel::firstNonOnlyReadableLayerIndex() const
   return -1;
 }
 
+QgsMapLayer *LayersModel::activeLayer()
+{
+  return mLayers.at( activeIndex() );
```

Comment:
https://doc.qt.io/qt-5/qlist.html#at

is activeIndex() always in bounds of the mLayers? if yes, add Q_ASSERT, if not , add a condition to return nullptr